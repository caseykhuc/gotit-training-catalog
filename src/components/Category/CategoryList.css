.menu-link {
  text-align: center;
  padding: 14px;
  text-decoration: none!important;
  color: rgba(0, 0, 0, 0.54);
}

.menu-link:hover {
  color: black;
}

.active-link {
  color: black;
}