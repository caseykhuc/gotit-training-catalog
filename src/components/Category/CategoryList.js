import React from 'react';
import { connect } from 'react-redux';
import { fetchCategory } from '../../actions';
import { Container, Row, Col } from 'react-bootstrap';
import { getCategories } from '../../reducers';

class CategoryList extends React.Component {
  componentDidMount() {
    if (!this.props.categories.length) {
      //console.log('hmm k on');
      this.props.fetchCategory();
    }
  }

  renderCategoryList = () => {
    return this.props.categories.map((category) => (
      <Col key={category.id}>{category.name}</Col>
    ));
  };

  render() {
    console.log(this.props);
    return (
      <Container style={{ textAlign: 'center' }}>
        <Row>{this.renderCategoryList()}</Row>
      </Container>
    );
  }
}

const mapStateToProps = (state) => ({
  categories: getCategories(state),
});

export default connect(mapStateToProps, { fetchCategory })(CategoryList);
