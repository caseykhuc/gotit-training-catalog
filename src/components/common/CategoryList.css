.menu-item-wrapper {
  outline: none;
}

.menu-link {
  text-align: center;
  padding: 14px;
  text-decoration: none!important;
  color: rgba(0, 0, 0, 0.54);
}

.menu-link:hover, .menu-link:focus {
  color: black;
  outline: none;
}

.active-link {
  color: black;
}