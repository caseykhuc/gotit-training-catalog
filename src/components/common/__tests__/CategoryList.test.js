import React from 'react';
import { shallow } from 'enzyme';
import { CategoryList } from '../CategoryList';

describe('component/Category/CategoryList', () => {
  let props;
  let wrapper;

  const update = () => {
    wrapper.update();
  };
  const setup = () => {
    wrapper = shallow(<CategoryList {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      categories: [
        { name: 'Name', id: 1 },
        { name: 'Name 2', id: 2 },
      ],
      match: { params: { categoryId: '1' } },
      history: { push: jest.fn() },
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should direct to other category when selected', () => {
    setup();
    wrapper.instance().onSelect(2);
    expect(props.history.push).toHaveBeenCalledWith('/categories/2');
  })
})
