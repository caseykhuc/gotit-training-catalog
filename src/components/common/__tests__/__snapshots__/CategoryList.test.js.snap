// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Common/CategoryList should render correctly 1`] = `ShallowWrapper {}`;
